lib/api/api.go

```diff
@@ -1173,6 +1174,30 @@ func (s *service) getDBIgnores(w http.ResponseWriter, r *http.Request) {
 	})
 }
 
+func (s *service) getIgnoresFromDeletedFolder(w http.ResponseWriter, r *http.Request) {
+	qs := r.URL.Query()
+
+	path := qs.Get("path")
+	finalPath := path + "/.stignore"
+	if _, err := os.Stat(finalPath); !os.IsNotExist(err) {
```

Comment:
Don't `stat()` the file beforehand; even if this succeeds, the file may be gone by the time we get to `open()`. Instead just try to open and handle the error by returning from the handler.

Reply:
can you clarify what you mean by that?

Reply:
You can remove the whole outer `if ... os.Stat()` thing. Either the file exists, and then you don't need it, or it does exist, and then you get the same error from `ioutil.ReadFile`. If you want to "mask" the does-not-exist error as a successful empty return, then do the `os.IsNotExist()` check on *that* error.